### ISSUE TITLE
Is it possible to use a USB device as the camera?

### ISSUE THREAD

**Issue by tylerssn** [blackuy/react-native-twilio-video-webrtc#567]
### Steps to reproduce
1. Plug in USB camera; Test that USB camera works on Android using the app "USB Camera - Connect EasyCap or USB WebCam"
2. No known feature in Twilio Video WebRTC that will allow me to use a USB camera

### Expected behaviour
Twilio Video should detect USB cameras and make it available for use

### Actual behaviour
Twilio Video is only aware of front and back camera

Has anyone been able to pull this off?

**Comment by jahead** [comment]
yes, this library doesn't support it out of the box, but the moving interface to camera2 instead of camera1 will allow you bring support for external cameras not just front and back, you can hack that in quite quickly, just pass the external camera2capture to the localVideoTrack. 

it also depends on how the manufacture has built the android image, some images of android don't come with UVC support, but most do. 

We currently are using two external cameras and a 'front' camera. But we had to extend this library a bit

**Comment by jahead** [comment]
see 
https://github.com/modusconnect /
[react-native-twilio-video-webrtc](https://github.com/modusconnect/react-native-twilio-video-webrtc)

if you want see how some of it was done. its not well doco'd
